metrics.go

```diff
@@ -110,3 +113,42 @@ func WriteFDMetrics(w io.Writer) {
 func UnregisterMetric(name string) bool {
 	return defaultSet.UnregisterMetric(name)
 }
+
+// sort labels alphabeticaly by label name
+type byKey []string
+
+func (k byKey) Len() int           { return len(k) / 2 }
+func (k byKey) Less(i, j int) bool { return k[i*2] < k[j*2] }
+func (k byKey) Swap(i, j int) {
+	k[i*2], k[j*2] = k[j*2], k[i*2]
+	k[i*2+1], k[j*2+1] = k[j*2+1], k[i*2+1]
+}
+
+// BuildName used to combine metric with labels
+func BuildName(name string, labels ...string) string {
+	if len(labels)%2 == 1 {
+		panic(fmt.Errorf("BUG: odd number of label/value pairs: %q", labels))
+	}
+
+	sort.Sort(byKey(labels))
+
+	var b strings.Builder
+	_, _ = b.WriteString(name)
+	_, _ = b.WriteRune('{')
+	for idx := 0; idx < len(labels); idx += 2 {
+		if idx > 0 {
+			_, _ = b.WriteRune(',')
+		}
+		_, _ = b.WriteString(labels[idx])
+		_, _ = b.WriteString(`="`)
+		_, _ = b.WriteString(labels[idx+1])
```

Comment:
strconv.AppendQuote must be used here in order to properly escape double quotes in label values